odoo/addons/base/models/ir_model.py

```diff
@@ -1994,7 +1994,9 @@ def _process_end(self, modules):
                     # Note: A regular module update in a regular (non-openupgrade) Odoo instance will remove this xmlids
                     record = self.env[model].browse(res_id)
                     if record.exists():
-                        continue
+                        if model != 'ir.ui.view':  # delete views always
+                            continue
```

Comment:
Your commit message mentions "delete obsolete xmlids in case of views". But if I understand correctly, this code prevents the actual record (actions, menu items, record rules, access rules) from being deleted if they no longer occur in the module (except views).

Reply:
Can you have a look at what happened here: https://github.com/OCA/OpenUpgrade/pull/2011/commits/0ddea82b61cab21f6202b2560a748eae6991cd5f. I think we should reinstate the deletion of all records that was removed in this code.

Reply:
why?

Reply:
It seems to me that otherwise, data that was removed from the modules (actions, menu items, record rules, access rules) are not removed from the migrated database.

Reply:
In https://github.com/OCA/OpenUpgrade/commit/0ddea82b61cab21f6202b2560a748eae6991cd5f says "# Note: A regular module update in a regular (non-openupgrade) Odoo instance will remove this xmlids"

Reply:
I'm talking about the records. The stanza that was removed in https://github.com/OCA/OpenUpgrade/commit/0ddea82b61cab21f6202b2560a748eae6991cd5f puts a `continue` before the actual record is unlinked.

Reply:
I don't know what you are trying to say, sorry. This is confusing. It was already discussed in https://github.com/OCA/OpenUpgrade/pull/2011#issuecomment-559075224.

Reply:
Yes, it was discussed I believe precisely for the same reason, and I don't understand why the faulty code got through after all (well, I made a mistake when I approved).

Let's have a look at a simple case:

In Odoo 13, a menu item was removed: https://github.com/OCA/OpenUpgrade/blob/13.0/odoo/addons/base/migrations/13.0.1.3/openupgrade_analysis.txt#L153

This menu item in Odoo 12 was not in a noupdate statement: https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/wizard/base_language_install_views.xml#L39-L43

OpenUpgrade, like Odoo itself, depends on such records to be removed automatically when the module is upgraded. The code that does that was removed/disabled by you in https://github.com/OCA/OpenUpgrade/commit/0ddea82b61cab21f6202b2560a748eae6991cd5f.

That means, that on an upgraded database, the menu item is still present:
```
odoo12base=> select latest_version from ir_module_module where name = 'base';
 latest_version 
----------------
 13.0.1.3
(1 row)

odoo12base=> select module, name, noupdate, res_id from ir_model_data
where name = 'menu_view_base_language_install';
 module |              name               | noupdate | res_id 
--------+---------------------------------+----------+--------
 base   | menu_view_base_language_install | f        |     49
(1 row)

odoo12base=> select name from ir_ui_menu where id = 49;
        name        
--------------------
 Load a Translation
(1 row)
```

I believe this is wrong. Do you agree?

Reply:
Not really, because after the migration, you will run an "update all" in a regular (non-openupgrade) Odoo instance that will remove that menu item.

Reply:
Then we disagree. Why change this behaviour in OpenUpgrade 13.0? What are the advantages?

Reply:
Well, I am not hard on this. Just done as a request by @pedrobaeza . The reason is explained in this comment: https://github.com/OCA/OpenUpgrade/pull/2011#issuecomment-559071019

And yes, I made an exception for views, because if I didn't, then later in the "update all" some deprecated views are validated and break.

Reply:
@StefanRijnhart the goal behind this is double:

* When you concatenate several migrations (for example doing a migration from 12 to 14). There are a lot of records that can be reused in later version, and you don't lose possible extra data attached to that records. Not so frequent.
* And the main one: if there's something that disappears in core, but you restore it later in a custom module. This way, you don't lose that records and possible extra data.

Reply:
Indeed, if you leave the records around they break the consistency of the data. This goes for views as much as for other data types I'm sure.

@pedrobaeza I doubt you have a concrete case because this is a huge, experimental change in behaviour.

Reply:
I had the old `crm_phonecall` case linked by @MiquelRForgeFlow, but if you think it's so conflictive, let's revert it.

Reply:
I would very much appreciate that. We can always reintroduce this new approach when there is a good case in OpenUpgrade 13. If ou13 is more mature then we can estimate the consequences much better.

Reply:
done